src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -106,6 +106,9 @@ public Block[] Process(Hash256 newBranchStateRoot, List<Block> suggestedBlocks,
         bool notReadOnly = !options.ContainsFlag(ProcessingOptions.ReadOnlyChain);
         int blocksCount = suggestedBlocks.Count;
         Block[] processedBlocks = new Block[blocksCount];
+
+        // Main thread should only read from prewarm caches, not spend extra time updating them.
+        _stateProvider.ReadOnlyPreWarmCaches = BlockchainProcessor.IsMainProcessingThread;
```

Comment:
pass it with `_preBlockCache` in constructor. Would call it `populatePreBlockCache` instead.

Reply:
Changed